cmake/CMakeLists.txt

```diff
@@ -12,6 +12,8 @@ cmake_policy(SET CMP0091 NEW)
 if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.20")
   cmake_policy(SET CMP0117 NEW)
 endif()
+# Don't let cmake set a default value for CMAKE_CUDA_ARCHITECTURES
+cmake_policy(SET CMP0104 OLD)
```

Comment:
Can you test the build can succeed with empty CMAKE_CUDA_ARCHITECTURES?
In my tests I saw build failures when I cleared CMAKE_CUDA_ARCHITECTURES. 

-- Configuring done
CMake Error in CMakeLists.txt:
  CUDA_ARCHITECTURES is empty for target "onnxruntime_test_cuda_ops_lib".

 


CMake Error in CMakeLists.txt:
  CUDA_ARCHITECTURES is empty for target "onnxruntime_test_cuda_ops_lib".

 


CMake Error in CMakeLists.txt:
  CUDA_ARCHITECTURES is empty for target "onnxruntime_providers_cuda".

 


CMake Error in CMakeLists.txt:
  CUDA_ARCHITECTURES is empty for target "onnxruntime_providers_cuda".

Reply:
yes, it works if you set policy to old.
but i'm leaning towards not merging this PR for the reasons @snnn mentioned.